Library/Homebrew/dev-cmd/irb.rb

```diff
@@ -21,15 +24,20 @@ module Homebrew
   module_function
 
   def irb
-    if ARGV.include? "--examples"
+    args = Homebrew::CLI::Parser.new do
+      switch "--examples"
+      switch "--pry"
+    end.parse
+
+    if args.examples?
       puts "'v8'.f # => instance of the v8 formula"
       puts ":hub.f.installed?"
       puts ":lua.f.methods - 1.methods"
       puts ":mpd.f.recursive_dependencies.reject(&:installed?)"
       return
     end
 
-    if ARGV.pry?
+    if args.pry? || Utils::EnvVars.pry?
```

Comment:
Is it possible for `args.pry?` to also handle the environment variable somehow? Also, you can probably delete https://github.com/Homebrew/brew/blob/121780b41701d1d4857b65c4a3f0ecd511807f75/Library/Homebrew/extend/ARGV.rb#L271-L273 now (and in similar other PRs).

Reply:
For `args.OPTION?` to handle environment variables, would probably have to assume that env var naming has the naming convention `HOMEBREW_<OPTION>`

Reply:
Yeh, that would work. Additionally, we could allow them to be specified with the DSL e.g. `env: :pry` or similar to mean `HOMEBREW_PRY` can trigger it.

Reply:
Explicitly specifying in DSL seems better to me.